### ISSUE TITLE
VEHICLE_UNAVAILABLE after Update/Installation of 3.0.2

### ISSUE THREAD

**Issue by t-moe-3** [alandtse/tesla#344]
I have already tried to see if other have the same problem, which does not seem to be the case:

I had the Tesla Custom Integration successfully running for months now (2.4.4 most recently) and waited for a quite day to take the time to uninstall and install the new 3.0.x (3.0.2 in my case). I did this today:

Made a full HA backup
uninstalled the custom integration V2.4.4
Installed the most recent version V3.0.2
I generated new tokens using the Android Tesla Token app.
Setting up the integration using my email and the refresh token it gives me the error:

**Logger: custom_components.tesla_custom
Source: custom_components/tesla_custom/__init__.py:253
Integration: Tesla Custom Integration (documentation, issues)
First occurred: 10:43:21 (8 occurrences)
Last logged: 11:00:15

Unable to communicate with Tesla API: VEHICLE_UNAVAILABLE**

The car is awake and charging, I have access through the Tesla app.

When I use a wrong token it has an error during authentication, so I assume the actual refresh token is correct.

Can anybody give me a tip on how to tackle this?

THANKS!

**Comment by fabianluque** [comment]
Same here after updating, both my vehicles entities showing unavailable.

**Comment by withanhdammit** [comment]
Not sure if this is related, I was running well, my Tesla was hit and totaled.  I got my new Tesla this week, trying to add it results in this:


2022-11-13 21:30:12.632 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration tesla_custom which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2022-11-13 21:36:13.112 INFO (MainThread) [homeassistant.setup] Setting up tesla_custom
2022-11-13 21:36:13.113 INFO (MainThread) [homeassistant.setup] Setup of domain tesla_custom took 0.0 seconds
2022-11-13 21:39:56.112 ERROR (MainThread) [homeassistant.config_entries] Error setting up entry <withanHdammit> for tesla_custom
File "/config/custom_components/tesla_custom/__init__.py", line 216, in async_setup_entry

**Comment by rberger** [comment]
I just did the update to v3.0.2 via HACS and the Integration stopped working. All entities show as `unavailable`I don't see any option to reauthenticate the integration.

**Comment by withanhdammit** [comment]
Since I'm dealing with a new car, I wasn't worried about losing history.  So I deleted the integration, then deleted it from HACS.  Restarted HA.  Reinstalled to v2.4.4  No help, same error:

2022-11-13 21:55:26.126 INFO (MainThread) [custom_components.hacs] <Integration alandtse/tesla> Download of tesla_custom.zip completed
2022-11-13 21:55:26.128 INFO (SyncWorker_7) [homeassistant.loader] Loaded tesla_custom from custom_components.tesla_custom
2022-11-13 21:55:26.128 WARNING (SyncWorker_7) [homeassistant.loader] We found a custom integration tesla_custom which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2022-11-13 21:59:58.178 ERROR (MainThread) [homeassistant.config_entries] Error setting up entry <withanHdammit> for tesla_custom
File "/config/custom_components/tesla_custom/__init__.py", line 216, in async_setup_entry
